<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <div id="userName">
        
    </div>
    <div>
        <button>home</button>
        <button>appointments</button>
        <button>chat history</button>
        <button>about</button>
    </div>
    <div>
        <button>book an appointment</button>
    </div>
    <div>
        <button onclick="logout()">logout</button>
    </div>
</body>
    <script>
        function getName(){ {
            return fetch(`api/patient?for=session`)
                .then(response => {
                    if (!response.ok) {
                        throw new Error('Network response was not ok ' + response.statusText);
                    }
                    return response.json();
                })
                .then(data => {
                    console.log(data[0].PatientLa)
                    const LastName = data[0].PatientLastName
                    const firstName = data[0].PatientName
                    const userNameDiv = document.getElementById('userName')
                    userNameDiv.innerHTML = `${LastName}, ${firstName}`
                    return data; // Return the data to the caller
                })
                .catch(error => {
                    console.error('There has been a problem with your fetch operation:', error);
                    return []; // Return an empty array on error
                });
            }
        }
        getName()

        async function logout(){
            try {
                    const response = await fetch('/api/patient?for=logout', {
                    method: 'POST',
                    headers: {
                    'Content-Type': 'application/json',
                    },
                    body: JSON.stringify({}),
                    });
        
                    if (!response.ok) {
                    throw new Error('Network response was not ok');
                    }

                    const result = await response.json();
                    console.log(result.hasOwnProperty('customError'))
                    if (result.hasOwnProperty('customError')) {
                        alert(result.customError)
                        return 
                    }
                    
                    window.location.href = '/login'
                    // Handle the result (e.g., show a success message to the user)
            } catch (error) {
                    console.error('Error:', error);
                    // Handle errors (e.g., show an error message to the user)
            }
        };
        
    </script>
</html>